text/3553-cargo-sbom.md

```diff
@@ -0,0 +1,121 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The generation of SBOM information is controlled by Cargo's configuration. To enable SBOM generation, set the following configuration:
+
+```toml
+[build]
+sbom = true
+```
+
+Or use the environment variable `CARGO_BUILD_SBOM=true`.
+
+If enabled, an SBOM file will be placed next to each compiled artifact for `bin`, `staticlib`, `cdylib` crate types in the `target` directory with the name `<artifact>.cargo-sbom.json`. The SBOM will contain information about dependencies used to build the compiled artifact.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The SBOM file generated by Cargo is *not* intended as a final SBOM artifact, but rather a precursor. Post-processing tooling can use the information produced here as part of building a final SBOM.
+
+The SBOM file will be written to disk before `rustc` is executed for the each artifact. This enables [`RUSTC_WORKSPACE_WRAPPER`](https://doc.rust-lang.org/cargo/reference/config.html#buildrustc-workspace-wrapper) to point at a program that can utilize the SBOM file to embed information into the binary if desired. The environment variable `CARGO_SBOM_PATH` will be set to the full path to the SBOM file.
+
+## Format
+The format will use JSON, but the exact format is not specified in this RFC. Additional fields can be added as needed. The JSON will include a `format-version` in case breaking changes are necessary.
+
+### Resolved Dependency Tree
+The SBOM file will include the following information for each crate. Note that a crate refers to a [single build unit (library or executable)](https://doc.rust-lang.org/cargo/appendix/glossary.html#crate).
+A [package](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) may contain multiple crates.
+- Package ID
+- Name
+- Version
+- Source (registry / git / path etc.)
+- Checksum (if available)
+- Dependencies
+- Type (normal, build)
+- Activated features
+
+If a crate is used as both a normal dependency and a build dependency that is separately compiled, then separate entries will exist in the dependency tree with the correct activated features listed for each instance.
+
+Checksum is an optional field, since only crates from registries have checksums. If a checksum is needed for a crate that comes from a path dependency for example, it will be up to the post-processing tool to produce an appropriate value.
+
+If further information is needed (such as license), then the post-processing tool can use `cargo metadata` or another mechanism to find it.
+
+### Resolved build configuration
```

Comment:
We might also want to track what target platform the SBOM of the artifact is associated with.